src/software/ai/navigator/path_planner/enlsvg_path_planner.h

```diff
@@ -0,0 +1,152 @@
+     *
+     * @param p/gv               point to be converted
+     *
+     * @return EnlsvgPoint/Point converted point
+     */
+    EnlsvgPoint convertPointToEnlsvgPoint(const Point &p) const;
+    Point convertEnlsvgPointToPoint(const EnlsvgPoint &gv) const;
+
+    /**
+     * Converts the internal representation of a path to a Path
+     *
+     * @param p          implenetation's representation of a path
+     *
+     * @return Point    converted Thunderbots Path
+     */
+    std::optional<Path> convertEnlsvgPathToPath(const EnlsvgPath &p) const;
+
+    /**
+     * Creates the obstacles in the grid from the given obstacles as well as
+     * obstacle-izing regions from the edges of the grids based on the
+     * grid_boundary_margin_offset.
+     *
+     * @param obstacles                      a list of obstacles
+     * @param grid_boundary_margin_offset    an offset that represts the width of the
+     * region from the edges of the grid to consider as an obstacle. Has the same units as
+     * the resolution parameter in EnlsvgPathPlanner()
+     */
+    void createObstaclesInGrid(const std::vector<ObstaclePtr> &obstacles,
+                               double grid_boundary_margin_offset) const;
+
+    /**
+     * Returns true if the internal representation of a coordinate is within the limits of
+     * the grid.
+     *
+     * @param gv     a internal coordinate representation to check
+     *
+     * @return bool  true if the point is within the limits of the field, false otherwise
+     */
+    bool isCoordNavigable(const EnlsvgPoint &gv) const;
+
+    /**
+     * Returns the closest unblocked internal coordinate for a point. It may return the
+     * given point itself.
+     *
+     * @param ep an internal point to investigate
+     *
+     * @return std::optional<EnlsvgPoint>    returns std::nullopt if the entire field is
+     * blocked, or an internal grid coordinate representing a closest unblocked location
+     */
+    std::optional<EnlsvgPoint> findClosestUnblockedEnlsvgPoint(
+        const EnlsvgPoint &ep) const;
+
+    /*
+     * Returns true if a given internal coordinate is blocked, false otherwise.
+     *
+     * @param ep an internal point to check
+     *
+     * @return   true if the point is blocked in the internal grid and false otherwise.
+     * Also returns true if the given coordinate is out-of-bounds
+     */
+    bool isBlocked(const EnlsvgPoint &ep) const;
+
+    // analogous to the size of a grid cell
+    double resolution;
+
+    // can be arbitrarily thought of as the width of the grid on the x-axis
+    int num_grid_rows;
+    // can be arbitrarily thought of as the height of the grid on the y-axis
+    int num_grid_cols;
+
+    Point origin;
+
+    int max_navigable_y_enlsvg_point;
+    int max_navigable_x_enlsvg_point;
+
+    // Required internal data structures
+    std::unique_ptr<EnlsvgGrid> grid;
+    std::unique_ptr<const EnlsvgAlgorithm> algo;
+    std::unique_ptr<EnlsvgMemory> mem;
```

Comment:
nit: for readability sake, it would be nice if these were prefixed with `enlsvg_`

Reply:
fixed